homeassistant/components/template/__init__.py

```diff
@@ -47,6 +50,10 @@ async def _reload_config(call: Event | ServiceCall) -> None:
         if conf is None:
             return
 
+        # Save persistent state so that upon reload it can be retrieved.
+        _LOGGER.debug("Saving %s states for reload", DOMAIN)
+        await restore_state.RestoreStateData.async_save_persistent_states(hass)
+
```

Comment:
Why do we need to do this manually, it should be done automatically by `RestoreEntity` when the entity is removed.

Reply:
During my testing I found out that upon reload the latest state did not always exist upon reload. Adding this fixed that. Thinking maybe that upon removing an entity the saving of the state is just "scheduled" or put on the loop but not completed when the entity is then reloaded.

Reply:
Have not seen behavior anymore but will keep on testing.